src/bootstrap/src/utils/helpers.rs

```diff
@@ -52,9 +54,22 @@ pub fn exe(name: &str, target: TargetSelection) -> String {
     crate::utils::shared_helpers::exe(name, &target.triple)
 }
 
+/// On AIX, both static library and dynamic library end with '.a'. One is bigaf format,
+/// one is in xcoff format.
+fn is_aix_dylib(path: &Path) -> bool {
+    let mut buffer = [0u8; 2];
+    match File::open(path).and_then(|mut f| f.read_exact(&mut buffer)) {
```

Comment:
Reading the file magic doesn't really seem correct. On AIX, archive files which contain XCOFF members with the appropriate bits set should still be dylibs.